### ISSUE TITLE
GitHub's "Stars" page sorts by the amount of code

### ISSUE THREAD

**Issue by ZYSzys** [open-source-ideas/open-source-ideas#94]
## Project description

GitHub is a good place for some beginners where they can learn how to code or improve their code ability.But sometimes it's difficult to know how to start learning and where to start. So I think if the GitHub's "Stars" page can sort by the amount of code, maybe it's better for code beginners. Beginners (like me) can sort the repos by the amount of code from little to much and learn code step by step which makes learning more progressive.

## Relevant Technology

JavaScript, GitHub, Chrome Extension.

## Who is this for

Novice.

## Complexity and required time

_Please only tick off one box in each category by changing `[ ]` to `[x]`. The labels on the project will then be updated by the maintainers as soon as possible._

### Complexity

- `[ ]` Beginner - This project requires no or little prior knowledge of the technolog(y|ies) specified to contribute to the project
- `[x ]` Intermediate - The user should have some prior knowledge of the technolog(y|ies) to the point where they know how to use it, but not necessarily all the nooks and crannies of the technology
- `[ ]` Advanced - The project requires the user to have a good understanding of all components of the project to contribute

### Required time (ETA)

- `[x ]` Little work - A couple of days
- `[ ]` Medium work - A week or two
- `[ ]` Much work - The project will take more than a couple of weeks and serious planning is required

---

Share on Twitter/other social media if you will. Something like

```
New project on Open Source Ideas Initiative: [fill in short description] – [fill in link]
```

**Comment by franckstifler** [comment]
I think the number of commits is relevant than the amount of codes. With a commit at least if it is well done you can have an idea of implementation of a functionality; a project with less commits will then mean that it is possible to catch up with some efforts. Also with commits you have the possibility to navigate through the development of the project step by step.

**Comment by ZYSzys** [comment]
Yeah, I agree with you in some points. But as for those have similar standard commits' style, I think beginners would like to choice repo which has less codes to read and study deeply. The repos which have enormous codes makes newbie difficult to start.

On the other hand, sorting the starred repos by the number of commits is another awesome idea.

**Comment by BhanuPrakashNani** [comment]
I am very much interested in this project and want to be a part of it. I really like doing new things. I am searching for an interesting project like this because I wanted to learn about GitHub API. I thought it would be better if it is a practical learning.  I have started learning regarding the GitHub API and I am comfortable with Javascript. When I came across this project I felt this is what I needed. Please give me the opportunity to work on this project with you. I feel this project is a very good idea. It really would help the new users to easily identify the repositories. I seriously want to be a part of it. Thank you.

**Comment by darkLord19** [comment]
What do you mean by github "stars" page? Do you mean trending repos?

**Comment by BhanuPrakashNani** [comment]
It is the page where we have all our Starred repositories from our account. Starring a repository makes it easy to find again later. You can see all the repositories you have starred by going to your stars page.

**Comment by darkLord19** [comment]
Ohh yes that. Thanks for clarification. This sounds interesting.

**Comment by darshkpatel** [comment]
Aren't we already able to see the starred repo's ?

**Comment by ZYSzys** [comment]
But I don't know how to start clearly 🤕.Are there any suitable GitHub APIs for this ?

**Comment by janriemer** [comment]
@ZYSzys You have to dig a little in Github's API documentation.
They have, e.g. documentation on how to do several things with repositories etc:
https://developer.github.com/v3/repos/
However, I'm not sure whether they have an api that shows lines of code in a package.
Counting lines by yourself for every repo out there is probably not a good idea. :laughing:
